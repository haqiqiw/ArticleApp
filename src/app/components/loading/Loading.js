import React from 'react';
import {
  Content,
  Spinner
} from 'native-base';

const Loading = ({containerStyle, color}) => {
  return (
    <Content contentContainerStyle={containerStyle}>
      <Spinner color={color} />
    </Content>
  );
}

Loading.defaultProps = {
  color: '#324191'
}

export default Loading;