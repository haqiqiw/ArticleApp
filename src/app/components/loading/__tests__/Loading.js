import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Loading from '../Loading';

describe('Loading Component', () => {
  it('should render without issue', () => {
    const component = shallow(<Loading />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should pass color props should still work', () => {
    const component = shallow(<Loading color="grey" />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should apply container style in the loading', () => {
    const component = shallow(
      <Loading containerStyle={{ backgroundColor: 'red' }} />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});