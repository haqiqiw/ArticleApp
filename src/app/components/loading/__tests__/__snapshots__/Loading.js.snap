// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component should apply container style in the loading 1`] = `
<Styled(Content)
  contentContainerStyle={
    Object {
      "backgroundColor": "red",
    }
  }
>
  <Styled(Spinner)
    color="#324191"
  />
</Styled(Content)>
`;

exports[`Loading Component should pass color props should still work 1`] = `
<Styled(Content)>
  <Styled(Spinner)
    color="grey"
  />
</Styled(Content)>
`;

exports[`Loading Component should render without issue 1`] = `
<Styled(Content)>
  <Styled(Spinner)
    color="#324191"
  />
</Styled(Content)>
`;
